src/clusterfuzz/_internal/bot/tasks/utasks/progression_task.py

```diff
@@ -378,112 +525,115 @@ def find_fixed_range(testcase_id, job_type):
     # If the min and max revisions are one apart this is as much as we can
     # narrow the range.
     if max_index - min_index == 1:
+      # TODO(alhijazi): This cannot be done from an untrusted bot.
       _save_fixed_range(testcase_id, min_revision, max_revision,
                         testcase_file_path)
-      return
+      return uworker_io.UworkerOutput(testcase=testcase)
 
     # Occasionally, we get into this bad state. It seems to be related to test
     # cases with flaky stacks, but the exact cause is unknown.
     if max_index - min_index < 1:
-      testcase = data_handler.get_testcase_by_id(testcase_id)
-      testcase.fixed = 'NA'
-      testcase.open = False
-      message = ('Fixed testing errored out (min and max revisions '
-                 'are both %d)' % min_revision)
-      data_handler.update_progression_completion_metadata(
-          testcase, max_revision, message=message)
-
-      # Let the bisection service know about the NA status.
-      bisection.request_bisection(testcase)
-      return
+      return uworker_io.UworkerOutput(
+          testcase=testcase,
+          uworker_input=uworker_input,
+          progression_task_output=uworker_io.ProgressionTaskOutput(
+              min_revision=min_revision, max_revision=max_revision),
+          error=uworker_msg_pb2.ErrorType.PROGRESSION_BAD_STATE_MIN_MAX)
 
     # Test the middle revision of our range.
     middle_index = (min_index + max_index) // 2
     middle_revision = revision_list[middle_index]
 
     testcase = data_handler.get_testcase_by_id(testcase_id)
-    log_message = 'Testing r%d (current range %d:%d)' % (
-        middle_revision, min_revision, max_revision)
+    log_message = (f'Testing r{middle_revision}'
+                   f'(current range {min_revision}:{max_revision})')
     data_handler.update_testcase_comment(testcase, data_types.TaskState.WIP,
                                          log_message)
 
-    try:
-      result = _testcase_reproduces_in_revision(testcase, testcase_file_path,
-                                                job_type, middle_revision)
-    except errors.BadBuildError:
-      # Skip this revision.
-      del revision_list[middle_index]
-      max_index -= 1
-      continue
+    result, error = _testcase_reproduces_in_revision(
+        uworker_input, testcase, testcase_file_path, job_type, middle_revision)
+    if error is not None:
+      if error.error == uworker_msg_pb2.ErrorType.PROGRESSION_BAD_BUILD:
```

Comment:
I don't think that would work.
I think `uworker_output::error` (ErrorType) should be renamed to error_type to avoid confusion.
We need to first check if error (which is a uworker_output) is not null. In that case we need to check error.error whether it is a `PROGRESSION_BAD_BUILD`.
Did I misunderstand something?

Reply:
Youre right it doesn't work.
Feel free to make the error_type change in new PR.